### PR TITLE
Use lower casing for OMS agent and correct solution for container monitoring

### LINKED ISSUE #1983: omsAgent profile is wrongly cased - it should be omsagent all lowercase
AKS cluster created with monitoring enabled is broken due to mixed case used in the 'omsAgent' where all other part of the azure is expecting lower case 'omsagent' 

https://github.com/terraform-providers/terraform-provider-azurerm/blob/master/azurerm/data_source_kubernetes_cluster.go

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* azurerm_XXXXX

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Steps to Reproduce

1. `terraform apply`

### References

* #0000

### PR DESCRIPTION
**Pull request by metacpp**
This PR includes:
1. Fix the issue #1983 caused by camelCase issue of "omsAgent".
2. Use `ContainerInsights` as the correct solution for container monitoring.
3. Update test cases and docs.

(fixes #1983)

### CODE REVIEW

**Comment by metacpp** [comment]
@ganga1980 please take a look at it.

File: website/docs/r/log_analytics_solution.html.markdown [hunk 1]
```diff
@@ -35,15 +35,15 @@ resource "azurerm_log_analytics_workspace" "test" {
...
+  solution_name         = "ContainerInsights"
   location              = "${azurerm_resource_group.test.location}"
   resource_group_name   = "${azurerm_resource_group.test.name}"
   workspace_resource_id = "${azurerm_log_analytics_workspace.test.id}"
   workspace_name        = "${azurerm_log_analytics_workspace.test.name}"
 
   plan {
     publisher = "Microsoft"
-    product   = "OMSGallery/Containers"
+    product   = "OMSGallery/ContainerInsight"
```
Comment (tombuildsstuff) [inline website/docs/r/log_analytics_solution.html.markdown]: this should be ContainerInsights*

**Comment by metacpp** [comment]
Log Analytics Solution Tests passed:
image

AKS Tests passed:
image

**Comment by metacpp** [comment]
@katbyte thanks for the merge!

**Comment by River991** [comment]
Any Idea when this will be released?

**Comment by metacpp** [comment]
@River991 it's planned in v1.17.0, should be very soon.

**Comment by River991** [comment]
Thats great to hear, this is blocking my team at the moment, love to see how quickly things are fixed here

**Comment by River991** [comment]
That's great news, would really help me out to have logging

**Pull request merged**